pyphare/pyphare/pharein/simulation.py

```diff
@@ -314,24 +369,36 @@ def check_refinement(**kwargs):
 
 
 
+def check_optional_keywords(**kwargs):
+    extra = []
+
+    if check_refinement(**kwargs) != "boxes":
+        extra += ['max_nbr_levels']
+
+    return extra
+
 
 # ------------------------------------------------------------------------------
 
 def checker(func):
     def wrapper(simulation_object, **kwargs):
         accepted_keywords = ['domain_size', 'cells', 'dl', 'particle_pusher', 'final_time',
                              'time_step', 'time_step_nbr', 'layout', 'interp_order', 'origin',
-                             'boundary_types', 'refined_particle_nbr', 'path',
-                             'diag_export_format', 'max_nbr_levels', 'refinement_boxes','refinement',
+                             'boundary_types', 'refined_particle_nbr', 'path', 'nesting_buffer',
```

Comment:
we probably need a check_nesting_buffer ?

Reply:
so that, if it's set, we make sure it's >= 0 ?

Reply:
we probably also want it to be less than there are cells in the patch domain... even half of it perhaps? something like that...

Reply:
I wasn't sure what to do here, but I think at least it can't be bigger than half the smallest_patch_size